src/drivers/mb12xx/mb12xx.cpp

```diff
@@ -270,16 +270,14 @@ MB12XX::init()
 
 	_class_instance = register_class_devname(RANGE_FINDER_BASE_DEVICE_PATH);
 
-	if (_class_instance == CLASS_DEVICE_PRIMARY) {
-		/* get a publish handle on the range finder topic */
-		struct distance_sensor_s ds_report = {};
+	/* get a publish handle on the range finder topic */
+	struct distance_sensor_s ds_report;
```

Comment:
This is missing the zero-initialiser braces, which will result in random memory content being published.